exp/api/v1beta1/azuremachinepoolmachine_types.go

```diff
@@ -27,6 +27,9 @@ import (
 const (
 	// AzureMachinePoolMachineFinalizer is used to ensure deletion of dependencies (nodes, infra).
 	AzureMachinePoolMachineFinalizer = "azuremachinepoolmachine.infrastructure.cluster.x-k8s.io"
+
+	// AzureMachinePoolMachineTestFinalizer is used to ensure deletion of dependencies during tests.
+	AzureMachinePoolMachineTestFinalizer = "azuremachinepoolmachine.infrastructure.cluster.x-k8s.io/test"
```

Comment:
Maybe it's worth defining this closer to the test itself? Could probably even be a local var right in the test.